File: Programmers/Python/Lv.2/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []
    function = deque(progresses)
    speeds = deque(speeds)
    
    while function :
        count = 0
        for idx in range(len(function)) :
            function[idx] = function[idx] + speeds[idx]

        while True :
            if function[0] < 100 :
                break
            if function[0] >= 100 :
                function.popleft()
                speeds.popleft()
                count += 1
            if len(function) == 0 :
                break
        if count > 0 :
            answer.append(count)
        
    
    
    return answer

File: Programmers/Javascript/Lv.2/기능개발.js
function solution(progresses, speeds) {
  let answer = [];

  while (progresses.length) {
    let count = 0;
    for (let i = 0; i < speeds.length; i++) {
      progresses[i] += speeds[i];
    }

    while (true) {
      if (progresses[0] >= 100) {
        progresses.shift();
        speeds.shift();
        count++;
      } else {
        break;
      }
    }
    if (count > 0) answer.push(count);
  }
  return answer;
}
